src/timeago-react.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { format, cancel, render } from 'timeago.js';
+
+/**
+ * Convert input to a valid datetime string of <time> tag
+ * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time
+ * @param input
+ * @returns datetime string
+ */
+const toDateTime = (input: string | number | Date): string => {
+  // let date: Date = new Date();
+  // if (input instanceof Date) {
+  //   date = input;
+  //   //@ts-ignore
+  // } else if (!isNaN(input) || /^\d+$/.test(input)) {
+  //   //@ts-ignore
+  //   date = new Date(parseInt(input));
+  // } else {
+  //   date = new Date(input);
+  // }
+
+  // try {
+  //   return date.toISOString();
+  // } catch (e) {
+  //   console.error('invalid datetime');
+  //   return '';
+  // }
+
+  return '' + (input instanceof Date ? input.getTime() : input);
+};
+
+export interface TimeAgoProps {
+  readonly datetime: string | number | Date; // date to be formatted
+  readonly live?: boolean; // real time render.
+  readonly relativeDate?: string | number | Date; // relative to which date
+  readonly minInterval?: number;
```

Comment:
combine relativeDate and minInterval as `opts`.

Reply:
Done!

One more request. Could you export the interfaces defined in timeago.js? So that others can use them directly instead of declaring them again.

Reply:
Very good advice, are you interested in submitting a pr?

Reply:
I found I could get the types from `lib/interfaces`, so I use them directly now. But meanwhile, I still made a PR for timeago.js in order to export interfaces directly from index.ts.